index.bs

```diff
@@ -310,9 +310,12 @@ is the rectangle we'll use to check against the targets.
 
 <dl class=switch>
 	<dt>If the <a>intersection root</a> is the <a>implicit root</a>,
-	<dd>it's the viewport's size.
+	<dd>it's treated as if the root were the <a>top-level browsing context</a>'s {{document}}, according to the following rule for {{document}}.
 
-	<dt>If the <a>intersection root</a> has an overflow clip,
+	<dt>If the <a>intersection root</a> is a {{document}},
+	<dd>it's the size of the {{document}}'s viewport.
```

Comment:
Where is "the document's viewport" defined?  Is it defined for documents that are not in a browsing context or are not fully active?

Reply:
This is sort of intentionally vague. The existing spec says this:

"If the intersection root is the implicit root, it [the root intersection rectangle] is the viewport’s size."

Elsewhere it mentions "the viewport of the document" and "viewport of the nested browsing context". It also says this in a highlighted note:

"Root intersection rectangle is not affected by pinch zoom and will report the unadjusted viewport, consistent with the intent of pinch zooming (to act like a magnifying glass and NOT change layout.)"

However, all of that text is likely to change with the resolution of this issue:

https://github.com/w3c/IntersectionObserver/issues/95

The intention here is to use "viewport" in a way that is consistent with existing usage in the spec, until the above issue is resolved.

Reply:
I understand all that, but the implicit root, being the active document of a toplevel browsing context, always has at least _some_ concept of a viewport.  If there's some random document involved (e.g. an XHR responseXML document, or a document from an iframe that is now removed from the DOM or whatever), there may be no concept of a viewport at all.  #95 does not cover that, as far as I can tell.

Reply:
The only spec language I could find that defines browser viewport is this unhelpful bit:

https://drafts.csswg.org/css2/visuren.html#viewport

But I'm not sure it's necessary to determine what the browser viewport of a detached document is, since the "Update the rendering" steps in the HTML event loop -- which is where IntersectionObserver runs -- only applies to fully active (connected) documents:

https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-processing-model

Reply:
Aha.  It might be good in the loop to assert that all documents we encounter are fully active.  Those might still not have a sane viewport (e.g. if their embedding iframe is display:none), but at least it cuts down on the cases that need to be considered.

Reply:
OK, I added some language there to clarify that the root document must be fully active at this step in processing.

PTAL

Reply:
This looks reasonable to me, thank you.

Reply:
@bzbarsky Would you mind clicking the "approved" button...